File: 3215-matrix-similarity-after-cyclic-shifts/matrix-similarity-after-cyclic-shifts.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {boolean}
 */
var areSimilar = function (mat, k) {
    const m = mat.length
    const n = mat[0].length
    k = k % n
    if (k === 0) {
        return true
    }
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            const currIdx = j
            let finalIdx;
            if (i % 2 === 0) {
                finalIdx = (j + k) % n
            } else {
                finalIdx = (j - k + n) % n
            }
            if (mat[i][currIdx] !== mat[i][finalIdx]) {
                return false
            }
        }
    }
    return true
};

File: 3215-matrix-similarity-after-cyclic-shifts/matrix-similarity-after-cyclic-shifts.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        m = len(mat)
        n = len(mat[0])
        k = k % n

        if k == 0:
            return True

        for i in range(m):
            for j in range(n):
                currIdx = j
                if i % 2 == 0:
                    finalIdx = (j + k) % n
                else:
                    finalIdx = (j - k + n) % n

                if mat[i][currIdx] != mat[i][finalIdx]:
                    return False
        return True
